backend/src/game/game.gateway.ts

```diff
@@ -0,0 +1,159 @@
+import {
+	SubscribeMessage,
+	MessageBody,
+	WebSocketGateway,
+	WebSocketServer,
+	OnGatewayInit,
+	OnGatewayConnection,
+	OnGatewayDisconnect,
+  } from '@nestjs/websockets';
+import { Logger } from '@nestjs/common';
+import { Socket, Namespace } from 'socket.io';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserService } from 'src/user/user.service';
+import { GameService } from './game.service';
+import { User, Match } from '@prisma/client'
+
+@WebSocketGateway({
+	namespace: 'game',
+	cors: {
+	  origin: 'http://localhost:3000', // Update with your client's origin
+	  methods: ['GET', 'POST'],
+	  transports: ['websocket'],
+	  credentials: true,
+	},
+})
+export class GameGateway
+implements OnGatewayInit, OnGatewayConnection, OnGatewayDisconnect
+{
+	// private paddleRightY: number = 250;
+	// private paddleLeftY: number = 250;
+	constructor(
+        private prisma: PrismaService,
+        private readonly userService: UserService,
+		private readonly gameService: GameService
+    ) {}
+	@WebSocketServer() server: Namespace;
+	private logger: Logger = new Logger('GameGateway');
+
+	// @SubscribeMessage('message')
+	// handleMessage(@MessageBody() data: { sender: string; message: string }) {
+	//   this.logger.log(`Message received from ${data.sender}: ${data.message}`);
+	//   // Broadcast the message to all connected clients
+	//   this.server.emit('message', data);
+	// }
+
+	@SubscribeMessage('createNewGame')
+	async handleNewGame(client: any, socketID: string) {
+		const newGame: Match = await this.gameService.newGame(socketID)
+		this.server.emit('newGame', {
+			game: newGame,
```

Comment:
Als je niet het hele object 'game' meestuurt over de websocket, maar alleen aan de frontend laat weten dat er een nieuwe game is (je zou bijvoorbeeld alleen ('newGame', game.ID) kunnen sturen). En dat de frontend daarna een GET request maakt voor de nieuwe game. Dan kan je beter beveiligen dat de juiste persoon alleen toegang heeft tot deze data, want de GET request kan je met de token beveiligen.. Ik probeer dat nu ook te implementeren in mijn code en vind dat eigenlijk wel een hele elegante manier. Het is iets meer werk, maar wel heel secure en ook overzichtelijk en makkelijk als je nog meer code gaat schrijven (scalable)